src/main/java/it/pagopa/pn/service/desk/service/impl/NotificationAndMessageServiceImpl.java

```diff
@@ -0,0 +1,168 @@
+                            return Mono.error(new PnGenericException(ERROR_ON_DELIVERY_CLIENT, exception.getMessage()));
+                        })
+                )
+                .flatMapMany(internalIdAndSentNotificationDto -> {
+                    if (!isNotificationCancelled(internalIdAndSentNotificationDto.getT2(), iun)) {
+                        response.setDocumentsAvailable(true);
+                        return Flux.fromIterable(internalIdAndSentNotificationDto.getT2().getDocuments())
+                                .flatMap(notificationDocumentDto ->
+                                        pnDeliveryClient.getReceivedNotificationDocumentPrivate(iun, Integer.parseInt(notificationDocumentDto.getDocIdx()), internalIdAndSentNotificationDto.getT1(), null)
+                                                .switchIfEmpty(Mono.empty())
+                                                .onErrorResume(exception -> {
+                                                    log.error("errorReason = {}, An error occurred while calling the service to obtain sent notifications", exception.getMessage());
+                                                    return Mono.error(new PnGenericException(ERROR_ON_DELIVERY_CLIENT, exception.getMessage()));
+                                                }))
+                                .map(notificationAttachmentDownloadMetadataResponseDto -> {
+                                    documentsSize.set(documentsSize.get() + notificationAttachmentDownloadMetadataResponseDto.getContentLength());
+                                    return NotificationAndMessageMapper.getDocument(notificationAttachmentDownloadMetadataResponseDto);
+                                });
+                    }
+                    return Flux.empty();
+                })
+                .switchIfEmpty(Mono.empty())
+                .collectList()
+                .map(documentList -> {
+                    response.setDocuments(documentList);
+                    response.setTotalSize(documentsSize.get());
+                    return response;
+                });
+    }
+
+    private boolean isNotificationCancelled(SentNotificationDto sentNotificationDto, String iun) {
+        AtomicBoolean cancellationTimelineIsPresent = new AtomicBoolean();
+        return pnDeliveryPushClient.getNotificationHistory(iun, sentNotificationDto.getRecipients().size(), sentNotificationDto.getSentAt())
+                .switchIfEmpty(Mono.empty())
+                .onErrorResume(exception -> {
+                    log.error("errorReason = {}, An error occurred while call service for obtain notification history", exception.getMessage());
+                    return Mono.error(new PnGenericException(ERROR_ON_DELIVERY_CLIENT, exception.getMessage()));
+                }).map(notificationHistoryResponseDto -> {
```

Comment:
wrappiamo tutto quello che c'è nel metodo map in un metodo

Reply:
come richiesto, abbiamo wrappato il contenuto del metodo map in un altro metodo privato